### PR TITLE
Feat GitHub oauth 219

### PR DESCRIPTION
**Pull request by BastinJafari**
Basic login implementation

### CODE REVIEW

**Comment by deepankarm** [comment]
@BastinJafari I'll create a new api gateway endpoint with github authorizer for this testing to unblock you tomorrow.

File: src/views/LogIn.tsx [hunk 1]
```diff
@@ -1,42 +1,46 @@
...
-          </Col>
-        </Row>
-      </Container>
-    );
-  };
+function Login() {
+  const client_id = process.env.REACT_APP_CLIENT_ID
+
+  useEffect(() => {
+    const parsed = queryString.parse(window.location.search)
```
Comment (Roshanjossey) [inline src/views/LogIn.tsx:13]:
```suggestion
    const queryParams = queryString.parse(window.location.search)
```

File: src/services/github.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import axios from "axios"
+import store from "../redux"
+import { setUser, showError } from "../redux/global/global.actions"
+import logger from "../logger"
+import { User } from "../redux/global/global.types"
+
+export function loginGithub(code: string) {
```
Comment (Roshanjossey) [inline src/services/github.ts:7]: Could you write unit tests for these functions

**Review by Roshanjossey — CHANGES_REQUESTED** [review]

**Review by imsergiy — APPROVED** [review]

**Review by Roshanjossey — APPROVED** [review]

File: src/redux/global/global.types.ts [hunk 3]
```diff
@@ -69,23 +71,25 @@ export type TNavItem = {
...
-  provider: string
-  username: string
-  _json: string
-  _raw: string
+  nodeId?: string
+  profileUrl?: string
+  provider?: string
+  _json?: { [userAttribute: string]: string }
+  _raw?: string
+  photos?: [{ value: string }]
```
[the remark this answers is not in the document]
Comment (BastinJafari) [reply src/redux/global/global.types.ts:91]: Actually not!
Removed it

**Pull request merged**